app/dashboard/date-picker/step-btn.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCaretDown } from '@fortawesome/free-solid-svg-icons/faCaretDown';
+import { twentyFourHoursInMilliseconds } from '@/app/utils/constants';
+
+interface StepBtnProps {
+  increment: boolean;
+  startDate: Date;
+  minDate: Date;
+  maxDate: Date;
+  onChange: (date: Date | null) => void;
+}
+
+const StepBtn = ({
+  increment,
+  startDate,
+  minDate,
+  maxDate,
+  onChange,
+}: StepBtnProps) => {
+  const isDateOutOfBound = (date: Date): boolean => {
+    const dateInDay = new Date(date.toDateString()).getTime();
+    const minDateInDay = new Date(minDate.toDateString()).getTime();
+    const maxDateInDay = new Date(maxDate.toDateString()).getTime();
+
+    return dateInDay < minDateInDay || dateInDay > maxDateInDay;
+  };
+
+  const diff = increment
+    ? twentyFourHoursInMilliseconds
+    : -twentyFourHoursInMilliseconds;
+  const newDate = new Date(startDate.getTime() + diff);
+  const disabled = isDateOutOfBound(newDate);
```

Comment:
Optional: wrap these snippets into a `useMemo` hook to memoize the `newDate` and `disabled` state (can be optionally renamed to `isDisabled`) with the following dependecies list for better performance:
- `startDate`
- `increment`